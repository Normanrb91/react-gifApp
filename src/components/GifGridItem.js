import React from 'react';
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {
    return (
        <div className="card text-center">
            <div class="card-body ">
                <img src={url} alt={title} />
                <p className="mt-3">{title}</p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
