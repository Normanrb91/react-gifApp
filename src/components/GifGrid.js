import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {


    const {data:images, loading} = useFetchGifs(category);

    return(
        <>
        <h2 c>{category}</h2>

            {loading && 

                <div class="spinner-border text-success" role="status">
                    <span class="visually-hidden">Espere...</span>
                </div>
            }

            <div className="card-grid mb-2">
            {
                images.map(img =>(
                    <GifGridItem 
                         key={img.id}
                         {... img}
                    />
                ))
            }

        </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
