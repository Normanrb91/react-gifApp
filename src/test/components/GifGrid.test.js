import React from 'react';
import {shallow} from 'enzyme';
import {GifGrid} from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente <GridGif />', () => {
    
    const category = 'titulo';

    test('Debe mostrar el componente correctamente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar cuando se cargan las imagens useFetchGifs ', () => {

        const gifs = [{
            id: 'abc123',
            title: 'Norman prueba cosas',
            url: 'https://pruebas.com'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    

})
