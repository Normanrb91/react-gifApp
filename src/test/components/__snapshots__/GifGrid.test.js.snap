// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GridGif /> Debe mostrar el componente correctamente  1`] = `
<Fragment>
  <h3>
    titulo
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
